src/App/Pages/Vault/BaseCipherViewModel.cs

```diff
@@ -0,0 +1,74 @@
+﻿using System;
+using System.Threading.Tasks;
+using Bit.App.Abstractions;
+using Bit.App.Resources;
+using Bit.Core.Abstractions;
+using Bit.Core.Exceptions;
+using Bit.Core.Models.View;
+using Bit.Core.Utilities;
+using Xamarin.CommunityToolkit.ObjectModel;
+
+namespace Bit.App.Pages
+{
+    public abstract class BaseCipherViewModel : BaseViewModel
+    {
+        private readonly IAuditService _auditService;
+        protected readonly IDeviceActionService _deviceActionService;
+        protected readonly ILogger _logger;
+        protected readonly IPlatformUtilsService _platformUtilsService;
+
+        public abstract CipherView Cipher { get; set; }
```

Comment:
Why is this `abstract`, if it's for the additional properties to raise property change just do that as an abstract and have the Cipher as full property here:
`protected abstract string[] AdditionalPropertiesToRaiseOnCipherChanged { get; }`